test/cmdlineTests.sh

```diff
@@ -141,9 +141,16 @@ function test_solc_behaviour()
 
     if [[ " ${solc_args[*]} " == *" --standard-json "* ]]
     then
-        sed -i.bak -e 's/{[^{]*Warning: This is a pre-release compiler version[^}]*},\{0,1\}//' "$stdout_path"
+python3 - <<EOF
+import re, sys
+json = open("$stdout_path", "r").read()
+json = re.sub(r"{[^{}]*Warning: This is a pre-release compiler version[^{}]*},?", "", json)
+json = re.sub(r"},\s*]", "}]", json)
+json = re.sub(r"\"errors\":\s*\[\s*\],?\s*","",json)
+json = re.sub("\n\\s+\n", "\n\n", json)
```

Comment:
Let's add some comments explaining what is being removed. For the message it's pretty self-explanatory, less so for the small tweaks after it:
```suggestion
json = re.sub(r"},\s*]", "}]", json)                  # },] -> }]
json = re.sub(r"\"errors\":\s*\[\s*\],?\s*","",json)  # Remove "errors" dict if it's not empty
json = re.sub("\n\\s+\n", "\n\n", json)               # Remove any leftover trailing whitespace
```